pkg/kubelet/cm/dra/state/claiminfo.go

```diff
@@ -14,77 +14,46 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package dra
+package state
 
 import (
 	"fmt"
-	"sync"
 
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/sets"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 )
 
 // claimInfo holds information required
 // to prepare and unprepare a resource claim.
-type claimInfo struct {
-	sync.RWMutex
-
-	// name of the DRA driver
-	driverName string
-
-	// claimUID is an UID of the resource claim
-	claimUID types.UID
-
-	// claimName is a name of the resource claim
-	claimName string
-
-	// namespace is a claim namespace
-	namespace string
-
-	// podUIDs is a set of pod UIDs that reference a resource
-	podUIDs sets.Set[string]
-
-	// cdiDevices is a list of CDI devices returned by the
-	// GRPC API call NodePrepareResource
-	cdiDevices []string
+type ClaimInfo struct {
+	ClaimInfoState
 
 	// annotations is a list of container annotations associated with
 	// a prepared resource
-	annotations []kubecontainer.Annotation
+	Annotations []kubecontainer.Annotation
```

Comment:
Would it make sense to include annotations into the ClaimInfoState?

Reply:
I don't think we want to checkpoint annotations. Especially since we are planning on removing them from the API soon.

Reply:
we wanted to keep them separately. The reason is that we don't want ClaimInfo  struct changes will effect the checkpoint file. We want to avoid migration between ClaimInfo  in old kubelet to ClaimInfo in newer kubelet. The checkpoint file store only the minimal information we need to restore ClaimInfo state. In the ClaimInfo state we can store additional information which may make our life easier in the code from perspective.   (e.g annotation)

Reply:
would it make sense to generate annotations instead of caching them?

Reply:
to me it make sense. What was the reasons for caching the annotation in the first place?

Reply:
It was done to avoid calling CDI api too often, but I don't think it creates a lot of overhead.
It only makes sense to generate annotations on the fly if we remove ClaimInfo struct, in my opinion.

Reply:
No that I am doing the full review I have a better picture of this. We should for sure not include this here (nor have any notion of annotations anywhere inside this package). As I've mentioned in other comments, we should have a wrapping `cache` type in the main `dra` package that extends the `state` type with additional metadata (such as this) as well as hangs other functions off of it (i.e. the ones we always had previously for the cache).